docs/quick_start.rst

```diff
@@ -0,0 +1,154 @@
+programmers to recall is that Fortran identifiers are case-insensitive.
+
+The global structs IREP generates are statically declared, so the size of all
+array types is given and fixed at compile time.
+
+Strings
+-------
+
+Character strings are stored in the IR as blank-delimited arrays, one
+character per array element.  String vectors are stored as two-dimensional
+arrays of characters.  Strings are null-terminated.  String length is
+currently limited to 512 characters.  It is usually best to access
+character strings using one of the convenience functions provided.
+If table ``t`` contains a scalar string ``ss`` and a vector of strings ``vs``:
+
+    // C++
+    std::string my_ss(IR_STR(t.ss));    // Access scalar string
+    std::string my_vs(IR_STR(t.vs[0])); // First element of vector string
+
+    ! Fortran
+    character(len=64) :: my_ss, my_vs
+    my_ss = trim(ir_fstr(t%ss))          ! Access scalar string
+    my_vs = trim(ir_fstr(t%vs(:,1)))     ! First element of vector string
+
+``IR_STR`` and ``ir_fstr`` basically depend on ``sizeof``, or its Fortran
+equivalent. So they can be used (only) with operands for which ``sizeof``
+will return a sensible value. Note that default values can be given only
+for scalar strings in the IR header files.
+
+
+Compilers
+---------
+
+Gfortran received a critical patch at version 4.7.4, that allowed
+it to properly read derived type components in namelist statements.
+That version or later is necessary for the IR to function correctly.
+Portland Group compilers 14.7 and 15.1 were tested, and appear to
+function correctly.  IBM bgxlf and bgxlc, version 14.1, were tested,
+and appear correct.  Intel ifort 14.0 was tested, and appears to give
+correct results.
+
+
+Callback functions
+------------------
+
+The IREP ``lua_cb_data`` struct has an integer component ``fref``, which
+nominally contains a Lua reference to the associated Lua callback
+function.  We overload the meaning of the possible fref values to cover
+several cases of interest::
+
+    Value of fref     Meaning
+    ---------------------------------------------------------------------
+    LUA_REFNIL        No entry was in the Lua input.
+    LUA_NOREF         Lua input is a constant function, number or vector.
+    <something else>  Lua input is a real Lua function
+
+Fref is statically initialized to ``LUA_REFNIL``.  If no actual
+Lua input for the given function name is found, this will therefore be
+the value of fref.
+
+If an input entry is found, we check the type.  If it is a number,
+the value of fref is set to ``LUA_NOREF``, and the number is stored in
+the structure's ``const_val`` component.  If the entry is instead a Lua
+function, we put the function in the Lua registry and store a reference
+to it in ``fref``.
+
+In particular, we can use the value of fref to infer whether or not
+any number or function was read:
+
+    fref == LUA_REFNIL ==> No entry of any kind was available;
```

Comment:
These lines look weird in the generated docs.
(See: https://irep.readthedocs.io/en/irep-refactor/quick_start.html#callback-functions)

Do they need a pair of extra spaces at the end of each line to enforce line breaks?
Or a way to indicate that it is a code / verbatim environment?

Reply:
should be `::` instead of `:` on the preceding line of text.  Fixed.